### PR TITLE
Supabase integration: 89

### PR DESCRIPTION
**Pull request by nishad-ayanworks**
- Worked on the supabase integration for sign up & sign in
- Worked on the route authentication for the frontend

### CODE REVIEW

**Review by tipusinghaw — CHANGES_REQUESTED** [review]

File: src/utils/check-session.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+export const checkUserSession = async (cookies: AstroCookies): Promise<boolean> => {
+    const sessionCookie = cookies.get("session").value;
+    console.log(`SessionCookie::`, sessionCookie);
+
+    if (!sessionCookie) {
+        return false
+    }
+
+    const { data: { user }, error } = await supabase.auth.getUser(sessionCookie);
+    console.log(`USRE::`, user);
```
[the remark this answers is not in the document]
Comment (nishad-ayanworks) [reply src/utils/check-session.ts]: Resolved

File: src/config/checkUserSession.ts
[the remark this answers is not in the document]
Comment (nishad-ayanworks) [reply src/config/checkUserSession.ts:1]: Resolved

File: src/components/Authentication/signUpUser-names.tsx [hunk 2]
```diff
@@ -248,7 +248,10 @@ const SignUpUser2 = () => {
 
 			{
 				currentComponent === 'password' && (
-					<SignUpUser3 />
+					<SignUpUser3
```
[the remark this answers is not in the document]
Comment (nishad-ayanworks) [reply src/components/Authentication/signUpUser-names.tsx]: Resolved

File: src/components/Authentication/signUpUser-names.tsx [hunk 3]
```diff
@@ -61,6 +61,7 @@ const SignUpUser2 = () => {
 			lastName: values.lastName
 		})
 		setContinuePasswordFlag(true)
+		setCurrentComponent('password')
```
[the remark this answers is not in the document]
Comment (nishad-ayanworks) [reply src/components/Authentication/signUpUser-names.tsx]: Resolved

File: src/components/Authentication/signUpUser-names.tsx [hunk 4]
```diff
@@ -3,18 +3,18 @@ import 'react-toastify/dist/ReactToastify.css';
...
 } from 'formik';
-import { sendVerificationMail } from '../../api/Auth.js';
-import { apiStatusCodes } from '../../config/CommonConstant.js';
-
-import type { AxiosError, AxiosResponse } from 'axios';
 import { useEffect, useState } from 'react';
-import SignUpUser3 from './signUpUser-passkey.js';
+
 import React from 'react';
+import SignUpUser3 from './signUpUser-passkey.js';
```
[the remark this answers is not in the document]
Comment (nishad-ayanworks) [reply src/components/Authentication/signUpUser-names.tsx]: Resolved

File: src/components/Authentication/SignUpUser.tsx [hunk 5]
```diff
@@ -3,24 +3,23 @@ import 'react-toastify/dist/ReactToastify.css';
...
 import { useEffect, useState } from 'react';
-import { addDeviceDetails, generateRegistrationOption, verifyRegistration } from '../../api/Fido.js';
-import { startRegistration } from '@simplewebauthn/browser';
-import type { IdeviceBody, RegistrationOptionInterface } from '../Profile/interfaces/index.js';
-import secureRandomPassword from 'secure-random-password';
-import SignUpUser2 from './signUpUser-names.js';
-import React from 'react';
 
+import React from 'react';
+import SignUpUser2 from './signUpUser-names.js';
```
[the remark this answers is not in the document]
Comment (nishad-ayanworks) [reply src/components/Authentication/SignUpUser.tsx]: Resolved

File: src/components/Authentication/SignInUser.tsx [hunk 6]
```diff
@@ -9,8 +11,8 @@ import {
 import { useEffect, useState } from 'react';
 
 import { Alert } from 'flowbite-react';
-import SignInUser2 from './SignInUser-passkey';
 import RegistrationSuccess from './RegistrationSuccess'
+import SignInUser2 from './SignInUser-passkey';
```
[the remark this answers is not in the document]
Comment (nishad-ayanworks) [reply src/components/Authentication/SignInUser.tsx:15]: Resolved

File: src/components/Authentication/SignInUser-password.tsx [hunk 7]
```diff
@@ -86,7 +88,19 @@ const SignInUser3 = (signInUserProps: SignInUser3Props) => {
 
 		if (data?.statusCode === apiStatusCodes.API_STATUS_SUCCESS) {
 			await setToLocalStorage(storageKeys.TOKEN, data?.data?.access_token)
-			getUserDetails(data?.data?.access_token)
+
```
[the remark this answers is not in the document]
Comment (nishad-ayanworks) [reply src/components/Authentication/SignInUser-password.tsx:79]: Ok. Will check future recreations

File: src/components/Authentication/SignUpUser.tsx [hunk 8]
```diff
@@ -249,41 +248,41 @@ const SignUpUser = () => {
 	}
 
 	const redirectLandingPage = () => {
-        window.location.href = '/'
-    }
-console.log("nextflag::::::", nextflag)
+		window.location.href = '/'
+	}
+	console.log("nextflag::::::", nextflag)
```
[the remark this answers is not in the document]
Comment (nishad-ayanworks) [reply src/components/Authentication/SignUpUser.tsx]: Resolved

**Review by tipusinghaw — APPROVED** [review]

**Pull request merged**